doc/release/1.16.0-notes.rst

```diff
@@ -47,6 +47,12 @@ Even when no elements needed to be drawn, ``np.random.randint`` and
 distribution. This has been fixed so that e.g.
 ``np.random.choice([], 0) == np.array([], dtype=float64)``.
 
+``linalg.qr`` now works with empty matrices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Previously, a ``LinAlgError`` would be raised when empty matrix ("flat"
+or "skinny") is passed in. This has been fixed so that outputs of
+appropriate shapes are returned for the various modes.
+
```

Comment:
I wonder if this and the similar note about `lstsq` for your related work in #11594 could be combined since they are both about handling empty matrices for linalg functions, but maybe that's a pain to coordinate between 2 separate PRs & could be done after I suppose.

Reply:
Separated them because for `lstsq`, I was working in a strange way (`site-packages`). For `qr` it was a more straightforward "change and run tests".